osg_configure/modules/ce_attributes.py

```diff
@@ -0,0 +1,105 @@
+# BATCH_SYSTEMS here is both the config sections for the batch systems
+# and the values in the OSG_BatchSystems attribute since they are
+# coincidentally the same. If they ever change, make a mapping.
+from configparser import ConfigParser
+from typing import Dict
+
+from osg_configure.modules import utilities, subcluster
+from osg_configure.modules.exceptions import SettingError
+
+BATCH_SYSTEMS_CASE_MAP = {
+    'condor': 'Condor',
+    'lsf': 'LSF',
+    'pbs': 'PBS',
+    'sge': 'SGE',
+    'slurm': 'SLURM',
+}
+BATCH_SYSTEMS = list(BATCH_SYSTEMS_CASE_MAP.values())
+
+
+def empty_if_blank(value: str) -> str:
+    return "" if utilities.blank(value) else value
+
+
+def get_resource_from_config(config: ConfigParser) -> str:
+    return utilities.classad_quote(
+        empty_if_blank(
+            config.get("Site Information", "resource", fallback="")
+        )
+    )
+
+
+def get_resource_group_from_config(config: ConfigParser) -> str:
+    return utilities.classad_quote(
+        empty_if_blank(
+            config.get("Site Information", "resource_group", fallback="")
+        )
+    )
+
+
+def get_batch_systems_from_config(config: ConfigParser) -> str:
+    batch_systems = []
```

Comment:
Should this be a set to avoid potential duplicates?

Reply:
Yeah, sure.